crates/bevy_color_blindness/src/lib.rs

```diff
@@ -0,0 +1,450 @@
+}
+
+/// Component to identify your main camera
+///
+/// Adding this component to a camera will set up the post-processing pipeline
+/// which simulates color blindness. This is done by changing the render target
+/// to be an image, and then using another camera to render that image.
+///
+/// Cameras with `ColorBlindnessCamera` will have [`UiCameraConfig`] inserted with
+/// `show_ui` set to `false`. This is to ensure that UI elements are not rendered twice.
+/// In most cases, you will want to render UI using the final post-processing camera.
+/// If for some reason this behavior is not desired, please open an issue.
+///
+/// [`UiCameraConfig`]: bevy_ui::entity::UiCameraConfig
+#[derive(Component)]
+pub struct ColorBlindnessCamera;
+
+/// sets the target for newly added `ColorBlindCamera`s
+fn set_camera_target(
+    mut commands: Commands,
+    mut query: Query<(Entity, &mut Camera), Added<ColorBlindnessCamera>>,
+    inner: Res<InternalResource>,
+) {
+    for (entity, mut camera) in query.iter_mut() {
+        camera.target = RenderTarget::Image(inner.image.clone());
+        commands
+            .entity(entity)
+            .insert(UiCameraConfig { show_ui: false });
+    }
+}
+
+/// updates the percentages in the post processing material when the `ColorBlindnessMode` changes in Params
+fn update_percentages(
+    params: Res<ColorBlindnessParams>,
+    inner: Res<InternalResource>,
+    mut materials: ResMut<Assets<PostProcessingMaterial>>,
+) {
+    if params.is_changed() {
+        let mut mat = materials.get_mut(&inner.post).unwrap();
+
+        let mode = if params.enable {
+            &params.mode
+        } else {
+            &ColorBlindnessMode::Normal
+        };
+
+        mat.percentages = mode.percentages();
+    }
+}
+
+/// internal resource which holds the handles
+struct InternalResource {
+    image: Handle<Image>,
+    post: Handle<PostProcessingMaterial>,
+}
+
+/// creates the image, the material, the final camera, and the whole post-processing pipeline
+///
+/// based on the post-processing example
+/// `https://github.com/bevyengine/bevy/blob/main/examples/shader/post_processing.rs`
+fn setup(
+    mut commands: Commands,
+    mut windows: ResMut<Windows>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut post_processing_materials: ResMut<Assets<PostProcessingMaterial>>,
+    mut images: ResMut<Assets<Image>>,
+    asset_server: Res<AssetServer>,
+    params: Res<ColorBlindnessParams>,
+) {
+    asset_server.watch_for_changes().unwrap();
+
+    let window = windows.get_primary_mut().unwrap();
+    let size = Extent3d {
+        width: window.physical_width(),
+        height: window.physical_height(),
+        ..Default::default()
+    };
+
+    // This is the texture that will be rendered to.
+    let mut image = Image {
+        texture_descriptor: TextureDescriptor {
+            label: None,
+            size,
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
```

Comment:
for wasm/android compatibility, this should use `bevy_default()` like https://github.com/bevyengine/bevy/pull/5601